### PR TITLE
Add spotless for formatting Java source files

### PR DESCRIPTION
**Pull request by mulugetam**
### Description

1. Adds spotless for Java code formatting ( `./gradlew spotlessJavaCheck`). 
2. Formats existing code. 
3. Updates the deprecated gradle construct `plugins.withId('java')`. 

### Issues Resolved
Resolves inconsistent code formatting. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@mulugetam we probably need to exclude some binary files (as GA checks fail), thank you.

**Comment by mulerm** [comment]
@reta Looks to me that `./gradlew check` failed because spotless was not applied to files in the `.github` directory. It should pass now. Not sure what binary files we should exclude.

**Comment by reta** [comment]
> @reta Looks to me that `./gradlew check` failed because spotless was not applied to files in the `.github` directory. It should pass now. Not sure what binary files we should exclude.

Oh I see, thanks @mulugetam , I just briefly looked into failure I thought there are some attempts to format binary files, non issue

File: build.gradle [hunk 1]
```diff
@@ -56,7 +77,7 @@ allprojects {
     version += "-SNAPSHOT"
   }
 
-  plugins.withId('java') {
+  'java' {
```
Comment (reta) [inline build.gradle:80]:
```suggestion
  java {
```

File: build.gradle [hunk 2]
```diff
@@ -210,3 +231,7 @@ task updateVersion {
         ant.replaceregexp(file:'build.gradle', match: '"opensearch.version", "\\d.*"', replace: '"opensearch.version", "' + newVersion.tokenize('-')[0] + '-SNAPSHOT"', flags:'g', byline:true)
     }
 }
+
+configurations.all {
```
Comment (reta) [inline build.gradle:235]: Where jackson is leaking?

Comment (mulerm) [reply build.gradle:235]: Was getting conflicting version numbers for jackson with `./gradlew assemble` but not any more (not sure why). Will remove.

File: build.gradle [hunk 3]
```diff
@@ -9,6 +9,25 @@
  * GitHub history for details.
  */
 
+/*
```
[the remark this answers is not in the document]
Comment (mulerm) [reply build.gradle:12]: https://github.com/opensearch-project/OpenSearch/blob/main/gradle/formatting.gradle has attribution to Elasticsearch.

**Comment by mulugetam** [comment]
@reta Fixed. Please review.

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
@mulugetam could you please apply backport to `2.x` manually? Thank you

**Pull request merged**